cassandra-migration/src/main/java/org/cognitor/cassandra/migration/Database.java

```diff
@@ -37,29 +43,56 @@
     /**
      * Statement used to create the table that manages the migrations.
      */
-    private static final String CREATE_MIGRATION_CF = "CREATE TABLE %s"
+    private static final String CREATE_MIGRATION_CF = "CREATE TABLE %s %s"
             + " (applied_successful boolean, version int, script_name varchar, script text,"
             + " executed_at timestamp, PRIMARY KEY (applied_successful, version))";
 
+    /**
+     * Statement used to create the table that manages the leader election on migrations.
+     */
+    private static final String CREATE_LEADER_CF = "CREATE TABLE %s %s"
+            + " (keyspace_name text, leader uuid, took_lead_at timestamp, PRIMARY KEY (keyspace_name))";
+
     /**
      * The query that retrieves current schema version
      */
     private static final String VERSION_QUERY =
             "select version from %s where applied_successful = True "
                     + "order by version desc limit 1";
 
+    /**
+     * The query that attempts to get the lead on schema migrations
+     */
+    private static final String TAKE_LEAD_QUERY =
+            "INSERT INTO %s (keyspace_name, leader, took_lead_at) VALUES (?, ?, dateOf(now())) IF NOT EXISTS USING TTL %s";
+
+    /**
+     * The query that releases the lead on schema migrations
+     */
+    private static final String RELEASE_LEAD_QUERY = "DELETE FROM %s where keyspace_name = ? IF leader = ?";
+
     /**
      * Error message that is thrown if there is an error during the migration
      */
     private static final String MIGRATION_ERROR_MSG = "Error during migration of script %s while executing '%s'";
 
+    private static final int LEAD_TTL = 300;
+
+    private static final int TAKE_LEAD_WAIT_TIME = 10000;
+
+    private final UUID instanceId = UUID.randomUUID();
```

Comment:
should this maybe configurable by having a setter? And should it maybe point to the hostname or ip address by default? As the UUID is never printed I have no chance to know which host has which UUID. So just by checking the database I have no way of figuring out which host actually has the lead. For this I have to check the logs of all the hosts. From a pure administration point of view that does not sound user friendly ;-)

Reply:
You could have several processes on the same machine that are competing for the lead. I'll add a column in the schema leader table to store the hostname or IP address of the leader.
Sounds good?

Reply:
Sounds great 👍